src/main/java/at/hannibal2/skyhanni/features/misc/ChestValue.kt

```diff
@@ -0,0 +1,114 @@
+import io.github.moulberry.notenoughupdates.util.GuiTextures.accessory_bag_overlay
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.ContainerChest
+import net.minecraft.item.ItemStack
+import net.minecraft.util.EnumChatFormatting
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import org.lwjgl.opengl.GL11
+import java.awt.Color
+import java.util.*
+
+
+class ChestValue {
+
+    private val config get() = SkyHanniMod.feature.misc
+
+    var data = mutableMapOf<Int, MutableList<ItemStack>>()
+
+    private var itemValueTotal = 0.0
+    private var itemEntryList: MutableList<Map.Entry<ItemStack, Double>> = mutableListOf()
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!config.estimatedChestValue) return
+
+        var value = 0.0
+        var valueMap: MutableMap<ItemStack, Double> = mutableMapOf()
+
+        for ((slot, item) in event.inventoryItems) {
+            value += EstimatedItemValue.getEstimatedItemPrice(item, mutableListOf()).first * item.stackSize
+            valueMap.addOrPut(item, EstimatedItemValue.getEstimatedItemPrice(item, mutableListOf()).first * item.stackSize)
+        }
+        itemValueTotal = value
+        var i = 0
+
+        itemEntryList.clear()
+        while (i<=3){
+            var highestValueEntry: Map.Entry<ItemStack, Double>? = null
+            var highestValue: Double? = null
+
+
+            for (entry in valueMap.entries) {
+                if (highestValue == null || entry.value > highestValue) {
+                    highestValueEntry = entry
+                    highestValue = entry.value
+                }
+            }
+            if (highestValueEntry != null) {
+                itemEntryList.add(highestValueEntry)
+                valueMap.remove(highestValueEntry.key)
+            }
+            i += 1
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    fun onInventoryRender(event: DrawScreenAfterEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!config.estimatedChestValue) return
+        val mc = Minecraft.getMinecraft()
+        val fontRenderer = mc.fontRendererObj
+        if (Minecraft.getMinecraft().currentScreen !is GuiChest) return
+        val gui = mc.currentScreen
+        val gui2: GuiChest = gui as GuiChest
+        val cc: ContainerChest = gui2.inventorySlots as ContainerChest
+        val guiName: String = cc.lowerChestInventory.displayName.unformattedText
+        if (guiName != "Large Chest" && guiName != "Chest") return
+        val accessorGui: AccessorGuiContainer? = gui as? AccessorGuiContainer
+        Minecraft.getMinecraft().textureManager.bindTexture(accessory_bag_overlay)
```

Comment:
We try to prepare SkyHanni for 1.20 and this includes a split off from NEU as well. Please try to prevent the usage of additional NEU code in SkyHanni for the future